### ISSUE TITLE
@room pills don't render correctly when sent using Markdown input

### ISSUE THREAD

**Issue by frebib** [element-hq/element-x-android#2966]
### Steps to reproduce

1. Use the non-RTE input field and send an @room mention
2. Note that the event doesn't send a formatted body, nor does the message bubble render the @room as a pill

#### What did you expect?
`@room` to render as a pill

#### What happened instead?
It rendered as plain text

### Will you send logs?

No

### Are you willing to provide a PR?

No

**Comment by bmarty** [comment]
Without formating, this event is sent:

```
  "content": {
    "body": "@room Test",
    "m.mentions": {
      "room": true
    },
    "msgtype": "m.text"
  },
```

and with formatting:

```
  "content": {
    "body": "@room test 2",
    "format": "org.matrix.custom.html",
    "formatted_body": "@room test 2",
    "m.mentions": {},
    "msgtype": "m.text"
  },
```
 and this is how this render on Web:

<img width="132" alt="image" src="https://github.com/element-hq/element-x-android/assets/3940906/8f628ea4-028d-4210-a354-bcd649b4e705">

and on EXA this is the opposite:

<img width="160" alt="image" src="https://github.com/element-hq/element-x-android/assets/3940906/d0213d71-a6c4-43a9-bac1-ee4ec4ecd949">

Also this [line](https://github.com/element-hq/element-x-android/blob/develop/features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/components/event/TimelineItemTextView.kt#L91) is crashing in debug build, since `mentionSpan.rawValue` is equal to `@room` and this is not a valid userId.

We have several problems to fix, not only a rendering problem.

**Comment by frebib** [comment]
Intentional mentions are working now, but the markdown input is still not sending a formatted body, so the room pill still doesn't render as a pill. EW does render a pill even without a formatted_body though. Not sure what the intention is here
```
  "content": {
    "msgtype": "m.text",
    "body": "@room ",
    "m.mentions": {
      "room": true
    }
  },
```

**Comment by jmartinesp** [comment]
> Intentional mentions are working now, but the markdown input is still not sending a formatted body, so the room pill still doesn't render as a pill.

Oh, true. I did all my tests with messages with both an `@room` mention and a explicit one so they were always formatted and I never found this issue. Let me check if I can fix this remaining case.

**Comment by jmartinesp** [comment]
This should now be fixed with #3201.
